src/Registration.js

```diff
@@ -0,0 +1,199 @@
+    this.focusFirstName()
+  }
+
+  checkValidation(data){
+    let error = false;
+    // frst name & last name cant be equal validation
+    if((data.firstName && data.firstName !== "") && (data.lastName && data.lastName !== "") && data.lastName == data.firstName){
+      error = true;
+      this.setState({
+        nameError:"First Name and Last name is cant be equal."
+      });
+    }else{
+      this.setState({nameError:""});
+    }
+
+    // first name validation
+    if(!data.firstName && data.firstName === ""){
+      error = true;
+      this.setState({
+        firstNameError:"First name is required."
+      });
+    }else{
+      this.setState({firstNameError:""});
+    }
+
+    // last name validation
+    if(!data.lastName && data.lastName === ""){
+      error = true;
+      this.setState({
+        lastNameError:"Last name is required."
+      });
+    }else{
+      this.setState({lastNameError:""});
+    }
+
+    // frst name & last name cant be equal validation
+    if((data.firstName && data.firstName !== "") && (data.lastName && data.lastName !== "") && data.lastName == data.firstName){
+      error = true;
+      this.setState({
+        nameError:"First Name and Last name is cant be equal."
+      });
+    }else{
+      this.setState({nameError:""});
+    }
+
+    // password validation
+    if(!data.password && data.password === ""){
+      error = true;
+      this.setState({
+        passwordError:"password is required."
+      });
+    }else{
+      this.setState({passwordError:""});
+    }
+
+    // last name validation
+    if(!data.confirmPassword && data.confirmPassword === ""){
+      error = true;
+      this.setState({
+        confirmPasswordError:"Confirm Password is required."
+      });
+    }else{
+      this.setState({confirmPasswordError:""});
+    }
+
+    // password match validation
+    if((data.password && data.password !== "") && (data.confirmPassword && data.confirmPassword !== "") && data.password != data.confirmPassword){
+      error = true;
+      this.setState({
+        matchPasswordError:"password not matched."
+      });
+    }else{
+      this.setState({matchPasswordError:""});
+    }
+
+    // contact no validation
+    if((data.number && data.number !== "") && data.number.length != "10"){
+      // console.log(data.number.length)
+      error = true;
+      this.setState({
+        numberError:"Contact number should have 10 digit."
+      });
+    }else{
+      this.setState({numberError:""});
+    }
+    return error;
+  }
+
+  saveAndContinue(e) {
+    // e.preventDefault()
+    // Get values via this.refs
+    var data = {
+      firstName     : this.refs.firstName.value,
+      lastName     : this.refs.lastName.value,
+      email     : this.refs.email.value,
+      password : this.refs.password.value,
+      confirmPassword : this.refs.confirmPassword.value,
+      number    : this.refs.number.value,
+    }
+    console.log(data)
+
+    if(this.checkValidation(data) === false){
```

Comment:
If `checkValidation` already returns a boolean, we could simple did a check like ```if(!this.checkValidation(data)) { ... }```